### ISSUE TITLE
About Multi-task training

### ISSUE THREAD

**Issue by ZhaZhaFon** [BUTSpeechFIT/speakerbeam#1]
hi, 
thanks for sharing.
I wonder if a **multi-task training (reconstruction loss + speaker loss)** was used in this repo? as suggested in the paper of TD-SpeakerBeam(_Improving Speaker Discrimination of Target Speech Extraction with Time-domain SpeakerBeam, ICASSP 2020_)

**Comment by zmolikova** [comment]
I apologize for the late response, the issue somehow slipped under my radar. In case this is still of interest, I leave the response below.

We did not use the speaker loss in this repo to keep it simple. Also, we did not observe much gain on the LibriMix dataset, which we use in this example. To add the speaker loss, it would be necessary to:
* Modify the `LibriMixInformed` `Dataset` to return the speaker index. The speaker-id can be simply parsed from the utterance id. This is in `src/datasets/librimi_informed.py`.
* Modify the `BaseEncoderMaskerDecoderInformed` model to add the classifier on top of extracted enroll embedding. This is in `src/models/base_models_informed.py`.
* Modify the `SystemInformed` to get the speaker index from the batch, the output of the classifier from the model, and pass this to the loss function. This is in `src/models/system.py`
* Create the loss function itself and pass it to the system in `egs/libri2mix/train.py`.

**Comment by ZhaZhaFon** [comment]
> 

Thanks a lot, you codes and reply are very helpful

**Comment by ZhaZhaFon** [comment]
Thanks a lot. Your codes and reply are very helpful.

**Comment by zmolikova** [comment]
Thanks and sorry again for the late reply. I am closing the issue, feel free to re-open if you have more questions.
